### ISSUE TITLE
Avoid building libyaml if already present

### ISSUE THREAD

**Issue by alsora** [ros2/libyaml_vendor#40]
This package currently always invokes `build_yaml()` without checking if yaml is already available.
https://github.com/ros2/libyaml_vendor/blob/master/CMakeLists.txt#L87

Why not adding something like
```
if(NOT yaml_FOUND OR "${yaml_VERSION}" VERSION_LESS 0.2.5)
  build_libyaml()
endif()
```

**Comment by hidmic** [comment]
To the very best of my knowledge, this package pins a `libyaml` version to ensure API/ABI stability. But @clalancette or @ahcorde may have a bit more context.

**Comment by alsora** [comment]
Through the `yaml_VERSION` CMake variable it should be possible to ensure arbitrary requirements on the version.
We could even require a very specific version of the library.

The problem of the current approach is that it may silently break systems that already depend on libyaml, by having two different versions of the library.

Unfortunately, it's not even possible to ignore these ROS 2 vendor packages even if a dependency is already present in the system, as they are marked as REQUIRED dependencies of many other packages.

**Comment by hidmic** [comment]
> The problem of the current approach is that it may silently break systems that already depend on libyaml, by having two different versions of the library.

Fair point. To fail the build if the library is present but versions do not match may be inconvenient in some cases, so perhaps we can emit a warning so that it doesn't go by unnoticed. WDYT?

**Comment by clalancette** [comment]
Fixed by #45, so closing.
